wavelength-plugins/emoticons.js

```diff
@@ -2,41 +2,45 @@
 Emoticon plugin
 This plugin allows you to use emoticons in both chat rooms (as long as they are enabled in the room) and private messages.
 */
-'use strict';
+"use strict";
 
-const fs = require('fs');
-let emoticons = {'feelsbd': 'http://i.imgur.com/TZvJ1lI.png'};
-let emoteRegex = new RegExp('feelsbd', 'g');
+const FS = require("../lib/fs.js");
+
+let emoticons = {"feelsbd": "http://i.imgur.com/TZvJ1lI.png"};
+let emoteRegex = new RegExp("feelsbd", "g");
 WL.ignoreEmotes = {};
 try {
-	WL.ignoreEmotes = JSON.parse(fs.readFileSync('config/ignoreemotes.json', 'utf8'));
+	WL.ignoreEmotes = JSON.parse(FS(`config/ignoreemotes.json`).readIfExistsSync());
 } catch (e) {}
 
 function loadEmoticons() {
 	try {
-		emoticons = JSON.parse(fs.readFileSync('config/emoticons.json', 'utf8'));
+		emoticons = JSON.parse(FS(`config/emoticons.json`).readIfExistsSync());
 		emoteRegex = [];
 		for (let emote in emoticons) {
 			emoteRegex.push(escapeRegExp(emote));
 		}
-		emoteRegex = new RegExp('(' + emoteRegex.join('|') + ')', 'g');
+		emoteRegex = new RegExp(`(${emoteRegex.join(`|`)})`, `g`);
```

Comment:
Prefer `emoteRegex.join('|')` within the escaped backticks to make it easier to read.